numpy/lib/tests/test_function_base.py

```diff
@@ -258,6 +259,11 @@ def test_basic(self):
         y6 = np.matrix(rand(5, 5))
         assert_array_equal(y6.mean(0), average(y6, 0))
 
+        y7 = np.array([decimal.Decimal(x) for x in range(10)])
+        w7 = np.array([decimal.Decimal(1) for _ in range(10)])
+        w7 /= w7.sum()
+        assert_almost_equal(y7.mean(0), average(y7, weights=w7)) 
```

Comment:
Test looks good, but would probably be better as a `test_object_dtype` method, rather than just making `test_basic` longer